modules/openapi-generator/src/main/resources/elixir/request_builder.ex.mustache

```diff
@@ -101,7 +101,7 @@ defmodule {{moduleName}}.RequestBuilder do
   end
 
   def add_param(request, :headers, key, value) do
-    Tesla.put_header(request, key, value)
+    Tesla.put_header(request, to_string(key), value)
```

Comment:
Given that Tesla headers (like Phoenix headers) are supposed to be all lowercase, we may want something a little more robust than just `Kernel.to_string/1`.

Reply:
Agreed, also because Http [header field names are case insensitive](https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive)
```elixir
key
|> to_String()
|> String.downcase()
```
Would this be sufficient, or do you mean something different by "robust" ?

Reply:
It’d be a start.

Reply:
Funny enough, this function call is actually wrong and `Tesla.put_header` must not be used here anyway. 

The function should look something like this:
```
  def add_param(request, :headers, key, value) do
    headers =
      request
      |> Map.get(:headers, [])
      |> List.keystore(key, 0, {key, value})

    Map.put(request, :headers, headers)
  end
```

Secondly, I wouldn't change cases as the server should/would do this to respect the case insensitivity

Reply:
#### TLDR

I promised to come back, only to now postpone 2 more weeks. Shall we leave this open, or shall i create new pr's and retire this PR ?
This is still a blocker for my release & i don't want to keep fork around and relase, so i'll definitely come back on this.

##### New PR proposal

I'd like to address these topics in separate or in one PR:
- Anytype support
- Header bugfix, using *camel_cased* `:atoms` not `:Atom` for headers
- Supply some generated usage  example in Readme.md using options. 

##### Tesla configurability

Separately I'd like to use _exposed_ middlewares in Tesla, because I think that is what Tesla's composability is about, instead of _creating_ and running the client in it's api functions, but we'll have to decide whether that is a custom generator, an alternative generator in this repo or whether we come to the conclusion that [req](https://github.com/wojtekmach/req) and tesla should be configurable settings of the same generator.

The Api implementation should just compose a list of middlewares, applied to a user supplied (and defaulted) `%Tesla.Env{}`
desetialisation e.t.c. should be implemented via middleware composition (one per field).
This way the lib user could also use non-spec ways of modifying the calls (authentication, session headers, caching e.t.c), while OFC the default is to run according to spec

Reply:
With respect to the Tesla configurability, I’d need to see more of what you think. We may be thinking similarly, but I have *absolutely* shifted my thinking on Tesla *away* from `use Tesla` with `plug` macros and instead using runtime client building. In my opinion, this item should *absolutely* be opened as a discussion / discussion ticket, because there are several modernizations to the Elixir code base that should be done beyond the work that I did earlier this year and may not have time to get to myself.